extensions/BugModal/web/new_bug.js

```diff
@@ -1,48 +1,89 @@
-$(document).ready(function() {
+var comp_desc = {}
+
+function load_products(query, callback) {
     bugzilla_ajax(
             {
                 url: 'rest/bug_modal/products'
             },
             function(data) {
-                $('#product').empty()
-                $('#product').append($('<option>', { value: 'Select Product', text: 'Select Product' }));
-                // populate select menus
-                $.each(data.products, function(key, value) {
-                    $('#product').append($('<option>', { value: value.name, text: value.name }));
-                });
+                callback(data.products);
             },
-            function() {}
+            function() {
+                callback();
```

Comment:
No need of that here, as the data is already loaded (or else it goes out through the bugzilla_ajax' error function).